core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -486,15 +485,15 @@ class ZkUtils(zkClientWrap: ZooKeeperClientWrapper,
   }
 
   def leaderAndIsrZkData(leaderAndIsr: LeaderAndIsr, controllerEpoch: Int): String = {
-    Json.encode(Map("version" -> 1, "leader" -> leaderAndIsr.leader, "leader_epoch" -> leaderAndIsr.leaderEpoch,
+    Json.legacyEncodeAsString(Map("version" -> 1, "leader" -> leaderAndIsr.leader, "leader_epoch" -> leaderAndIsr.leaderEpoch,
                     "controller_epoch" -> controllerEpoch, "isr" -> leaderAndIsr.isr))
   }
 
   /**
    * Get JSON partition to replica map from zookeeper.
    */
-  def replicaAssignmentZkData(map: Map[String, Seq[Int]]): String = {
-    Json.encode(Map("version" -> 1, "partitions" -> map))
+  def replicaAssignmentZkData(partitionMap: Map[Int, Seq[Int]]): String = {
```

Comment:
Let's not change this signature since it may break users and there's not much benefit (this is an internal class, but used by many other projects, so we are avoiding change as much as possible until we remove it).

Reply:
Agreed and fixed.